lib/src/scheduler.dart

```diff
@@ -15,36 +14,39 @@
 // limitations under the License.
 //
 ////////////////////////////////////////////////////////////////////////////////
-
 import 'dart:ffi';
 import 'dart:isolate';
-
 import 'native/realm_core.dart';
 
 final _receivePortFinalizer = Finalizer<RawReceivePort>((p) => p.close());
 final Scheduler scheduler = Scheduler._();
 
 class Scheduler {
   late final SchedulerHandle handle;
-  final RawReceivePort receivePort = RawReceivePort();
+  final RawReceivePort _receivePort = RawReceivePort();
 
-  Scheduler._() {
-    _receivePortFinalizer.attach(this, receivePort, detach: this);
+  int get nativePort => _receivePort.sendPort.nativePort;
 
-    receivePort.handler = (dynamic message) {
-      realmCore.invokeScheduler(handle);
+  Scheduler._() {
+    _receivePortFinalizer.attach(this, _receivePort, detach: this);
+
+    _receivePort.handler = (dynamic message) {
+      if (message is List) {
+        realmCore.loggerLogMessage(message[0] as int, message[1] as String);
```

Comment:
```suggestion
       final message_level = LevelExt.fromInt(level);
       Realm.logger.log(message_level, message);
```
I thought that we don't have access to `LevelExt` here, but since we don I suggest we remove the `realmCore.loggerLogMessage` function and do the logging right here. Since `realmCore.loggerLogMessage`  look like we are actually calling Core.

Reply:
You could just change the name of the function, instead of adding package dependency.

Reply:
I am not sure why do you think we will add a package dependency